src/main/java/me/botsko/prism/database/sql/SqlPrismDataSourceUpdater.java

```diff
@@ -108,4 +108,30 @@ public void v7_to_v8() {
             dataSource.handleDataSourceException(e);
         }
     }
+
+    @Override
+    public void v8_to_v9() {
+        // Declare query
+        String query = "";
+
+        // Prepare database
+        try (
+                Connection conn = dataSource.getConnection();
+                PreparedStatement st = conn.prepareStatement(query);
+        ) {
+            // Add action_id index
+            query = "ALTER TABLE `" + prefix + "data` ADD INDEX `action` (`action_id`)";
+            st.executeUpdate(query);
+
+            // Drop location index to re-create it
+            query = "ALTER TABLE `" + prefix + "data` DROP INDEX `location`";
+            st.executeUpdate(query);
+
+            // Re-create location index
+            query = "ALTER TABLE `" + prefix + "data` CREATE INDEX `location` (`world_id`, `x`, `z`, `y`)";
```

Comment:
I know it's the only way to do this, but this might need to come with a hefty warning when it's released. One of our servers we has 55M rows in the `data` table and will likely halt startup for quite some time. An unexpected long startup time could cause someone to abort/kill the server and start it again, without knowing this is intentional. Perhaps this should come with a big console warning?

This is all called in `onEnable()` (and on main thread) but if I recall correctly, this will all happen before the main thread watchdog starts.

Reply:
Oh, actually I was wrong, it's not called in-line within `onEnable()`:
1. `onEnable()` called during on start up
2. Async task is launched to establish DB connections.
3. Sync task is launched (`enabled()`) to perform upgrades.

After reading this, now I am a little concerned about potential start up issues with the Watchdog restarting the server. This will need testing on a large database.. or you can just put a 300s sleep in the `v8_to_v9()` method to simulate a long upgrade time.

Reply:
I believe you make some really valid points in terms of largers servers. It might be prudent to start Prism with the recorder paused, run the upgrader async in the background then have the upgrader unpause the recorder once it's done. Might require a new PR for it.

Reply:
I can look at moving the updater async...although I thought it already was
Db startup is definitely already handled async

Reply:
I can confirm the updater is run async .. that being said is this still valid
@addstar @lflare